### PR TITLE
Tab navigator

### PR DESCRIPTION
**Pull request by czystyl**
related to #1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +33,8 @@
     "react-navigation": "^3.0.5",
     "react-navigation-stack": "^1.0.5",
     "@react-navigation/core": "^3.0.1",
-    "@react-navigation/native": "^3.0.3"
+    "@react-navigation/native": "^3.0.3",
+    "react-navigation-tabs": "1.0.1"
```
Comment (grabbou) [inline package.json]: What is this "react-navigation-tabs" used for?

Comment (grabbou) [reply package.json]: Ok, got it.

File: src/ReactNavigation.re [hunk 2]
```diff
@@ -14,4 +14,9 @@ module Native = {
 module Stack = {
   [@bs.module "react-navigation-stack"]
   external stackView: ReasonReact.reactElement = "StackView";
+};
+
+module TabNavigator = {
+  [@bs.module "react-navigation"]
```
Comment (grabbou) [inline src/ReactNavigation.re]: We shouldn't be calling main react-navigation module, right?

**Comment by grabbou** [comment]
Now that this PR is open, I am cross-posting comment from another PR so that it doesn't get lost: https://github.com/callstackincubator/bs-react-navigation/pull/8#pullrequestreview-186080367

**Comment by czystyl** [comment]
@vnovick Do you use the refmt? I see that some part of the codebase is not formatted.

I started checking, and what I faced at the beginning is that we lose the standard animations. It could be caused by the fact that we render only one component but want to check it. I'll go deep into implementation later on. 

BTW. The tab navigator is working, and it's great! 🎉🎉🎉

ezgif com-resize 1

**Comment by vnovick** [comment]
@czystyl, Yeah. I use refmt. It's just It wasn't enabled because of formatOnSave setting in Vscode was false. I needed that for different project and forgot to turn on. 
We actually don't render one component. We pass screens config to createBottomTabNavigator so not sure why there are no animations

File: src/TabNavigator.re [hunk 3]
```diff
@@ -36,30 +36,19 @@ type screenOptions = {
...
   [@bs.optional]
-  tabBarButtonComponent:
-    ReasonReact.componentSpec(
-      ReasonReact.stateless,
-      ReasonReact.stateless,
-      ReasonReact.noRetainedProps,
-      ReasonReact.noRetainedProps,
-      ReasonReact.actionless,
-    ),
+  tabBarButtonComponent: ReasonReact.reactElement,
```
[the remark this answers is not in the document]
Comment (czystyl) [reply src/TabNavigator.re]: Would be great if a user can deliver `Reason.reactElement` and inside lib we can use `ReasonReact.wramReasonForJS` to transform this to `ReasonReact.reactClass`. What do you think?

File: src/TabNavigator.re [hunk 4]
```diff
@@ -36,30 +36,19 @@ type screenOptions = {
...
   [@bs.optional]
   tabBarAccessibilityLabel: string,
   [@bs.optional]
   tabBarTestID: string,
-  /* [@bs.optional]
-     tabBarOnPress: string */
+  [@bs.optional]
+  tabBarOnPress: string,
```
[the remark this answers is not in the document]
Comment (czystyl) [reply src/TabNavigator.re]: Oh, sorry. I'm going to back this comment.

**Pull request merged**